File: Demo/src/main/java/com/myscript/iink/demo/IInkApplication.java
// Copyright MyScript. All rights reserved.

package com.myscript.iink.demo;

import android.app.Application;
import android.content.Context;

import com.myscript.certificate.MyCertificate;
import com.myscript.iink.Configuration;
import com.myscript.iink.Engine;

public class IInkApplication extends Application
{
  private static Engine engine;
  private static Context context;

  @Override
  public void onCreate() {
    super.onCreate();
    context = this;
  }

  public static synchronized Engine getEngine()
  {
    if (engine == null)
    {
      engine = Engine.create(MyCertificate.getBytes());
    }

    String configDir = "zip://" + context.getPackageCodePath() + "!/assets/conf";
    Configuration config = engine.getConfiguration();
    config.setStringArray("configuration-manager.search-path", new String[]{configDir});
    config.setString("lang", "zh_CN");
    config.setBoolean("gesture.enable", false);

    return engine;
  }

}

File: Demo/src/main/java/com/myscript/iink/demo/IInkApplication.kt
// Copyright MyScript. All rights reserved.

package com.myscript.iink.demo

import android.app.Application
import android.content.Context
import android.util.Log
import com.myscript.certificate.MyCertificate
import com.myscript.iink.Engine

class IInkApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        context = this
    }

    companion object {
        private var engine: Engine? = null
        private var context: Context? = null

        @Synchronized
        fun getEngine(): Engine {
            if (engine == null) {
                engine = Engine.create(MyCertificate.getBytes())
            }

            val configDir = "zip://" + context!!.packageCodePath + "!/assets/conf"
            Log.d(Consts.TAG, "config Dir: $configDir")
            val config = engine!!.configuration
            config.setStringArray("configuration-manager.search-path", arrayOf(configDir))
            config.setString("lang", "zh_CN")
            config.setBoolean("gesture.enable", false)

            return engine as Engine
        }
    }

}
